py/mattak/backends/uproot/dataset.py

```diff
@@ -278,8 +349,18 @@ def wfs(self, calibrated : bool = False) -> numpy.ndarray:
             if len(wf_idxs): 
                 w[wf_idxs] = self._wfs['radiant_data[24][2048]'].array(entry_start=wf_start, entry_stop=wf_end, library='np')
 
-        # here we'd eventually handle calibration I think? 
-        if self.multiple: 
+        # calibration
+        if calibrated and self.cal_file:
+            cal_param = self.__unpack_cal_parameters(self.cal_file)
+            cal_residuals_v, cal_residuals_adc = self.__unpack_cal_residuals(self.cal_file)
+            kw = dict(entry_start = self.first, entry_stop = self.last)
+            radiantStartWindows = self._hds['trigger_info/trigger_info.radiant_info.start_windows[24][2]'].array(**kw, library='np')
+            starting_window = radiantStartWindows[0, :, 0]
+            w = numpy.array([self.__calibrate(ele, cal_param, cal_residuals_v, cal_residuals_adc,  starting_window) for ele in w])
```

Comment:
Ar you sure everything is correct here with `starting_windows`? It looks wrong to me but I might be wrong

Reply:
No you are right, it should be a different starting window per waveform so the first index should be the index of the waveform, not just 0